modules/table/src/test/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/OutgoingSnapshotTest.java

```diff
@@ -0,0 +1,436 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed.raft.snapshot.outgoing;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.ignite.hlc.HybridClock;
+import org.apache.ignite.internal.schema.ByteBufferRow;
+import org.apache.ignite.internal.storage.ReadResult;
+import org.apache.ignite.internal.storage.RowId;
+import org.apache.ignite.internal.table.distributed.TableMessagesFactory;
+import org.apache.ignite.internal.table.distributed.raft.snapshot.PartitionAccess;
+import org.apache.ignite.internal.table.distributed.raft.snapshot.PartitionKey;
+import org.apache.ignite.internal.table.distributed.raft.snapshot.message.SnapshotMvDataRequest;
+import org.apache.ignite.internal.table.distributed.raft.snapshot.message.SnapshotMvDataResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class OutgoingSnapshotTest {
+    @Mock
+    private PartitionAccess partitionAccess;
+
+    @Mock
+    private OutgoingSnapshotRegistry snapshotRegistry;
+
+    private OutgoingSnapshot snapshot;
+
+    private final TableMessagesFactory messagesFactory = new TableMessagesFactory();
+
+    private final RowId lowestRowId = RowId.lowestRowId(1);
+    private final RowId rowId1 = Objects.requireNonNull(lowestRowId.increment());
+    private final RowId rowId2 = Objects.requireNonNull(rowId1.increment());
+    private final RowId rowId3 = Objects.requireNonNull(rowId2.increment());
+
+    private final RowId rowIdOutOfOrder;
+
+    {
```

Comment:
Why don't you use setUp method?
Oh wait, you already have `@BeforeEach`, why did you split it in two sections then?

Reply:
I like to initialize everything that is possible with constructors or, at least, initializers.

Switched to `@BeforeEach`, this doesn't seem to be a big deal.

There are two of them because they are not dependent on each other, so it's more modular and so on.

Reply:
I don't know, it's a common practice to initialize stuff either as final fields or in BeforeEach/BeforeAll, why adding something else. Having 2 "constructors" if very confusing, there's nothing "modular" about it